hollow/src/main/java/com/netflix/hollow/api/consumer/metrics/AbstractRefreshMetricsListener.java

```diff
@@ -163,29 +173,36 @@ public void refreshFailed(long beforeVersion, long afterVersion, long requestedV
 
     @Override
     public void snapshotUpdateOccurred(HollowAPI refreshAPI, HollowReadStateEngine stateEngine, long version) {
-        trackCycleStartTime(version, stateEngine.getHeaderTags());
+        trackTimestampsFromHeaders(version, stateEngine.getHeaderTags());
     }
 
     @Override
     public void deltaUpdateOccurred(HollowAPI refreshAPI, HollowReadStateEngine stateEngine, long version) {
-        trackCycleStartTime(version, stateEngine.getHeaderTags());
+        trackTimestampsFromHeaders(version, stateEngine.getHeaderTags());
     }
 
     /**
-     * If the blob header contains producer cycle start time tag then save its value in version to cycle start time map
+     * If the blob header contains the timestamps like producer cycle start and announcement then save those values in
+     * the maps tracking version to cycle start time and version to announcement respectively.
      */
-    private void trackCycleStartTime(long version, Map<String, String> headers) {
+    private void trackTimestampsFromHeaders(long version, Map<String, String> headers) {
```

Comment:
Refactoring with this method signature should simplify the implementation
```
trackTimestampsFromHeaders(long version, Map<String, String> headers, String headerTag, Map<String, Long> timestampsMap)
```

Reply:
Thanks. Done.